helpers/campaignLimiting.js

```diff
@@ -0,0 +1,35 @@
+const BN = require('bn.js')
+const db = require('../db')
+
+async function getCampaignsEarningFrom (query) {
```

Comment:
This implementation also has an issue: this `getCampaignsEarningFrom` does not consider the filters that the main query has (e.g. campaign status), which means it may return campaigns that are in a status that we don't want to return

To fix this, I recommend you pass down the `mongoQuery` to `filterCampaignsForPublisher` and then just extend it by copying it and adding the `[mongoQueryKey]`

also I recommend you restructure the args to `filterCampaignsForPublisher`: there's no point to pass the entire `req.query`; you just need `limitForPublisher`